nncf/torch/dynamic_graph/wrappers.py

```diff
@@ -136,7 +135,7 @@ def wrapped(self, *args, **kwargs):
         from nncf.torch.dynamic_graph.patch_pytorch import unpatching_module_call
 
         # If called on a model compiled by torch dynamo, we unpatch torch operators and invoke original module call
-        if isinstance(self, OptimizedModule):
+        if "_torchdynamo_orig_callable" in self.forward.__dict__:
```

Comment:
It seems that in theory any model method can be compiled in isolation. The fix presented here won't work for this case. To deal with such problem, we could do something like this:
```
for attr in self.__dict__.values():
    if hasattr(attr, "__dict__") and "_torchdynamo_orig_callable" in attr.__dict__:
        return unpatching_module_call(self, *args, **kwargs)
```
Such logic should protect the inference from being patched in case any of the model's methods is compiled. This however will introduce some overhead during every patched module call. I propose to keep it as is for now until we stumble across an issue where this can actually help.